stellarphot/photometry/profiles.py

```diff
@@ -0,0 +1,305 @@
+import warnings
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+from astropy.nddata import Cutout2D
+from astropy.nddata.utils import NoOverlapError
+from astropy.utils import lazyproperty
+
+from photutils.profiles import RadialProfile, CurveOfGrowth
+from photutils.centroids import centroid_com, centroid_2dg
+
+from stellarphot.photometry import calculate_noise
+
+__all__ = ["find_center", "CenterAndProfile"]
+
+
+def find_center(image, center_guess, cutout_size=30, max_iters=10, match_limit=3):
+    """
+    Find the centroid of a star from an initial guess of its position. Originally
+    written to find star from a mouse click.
+
+    Parameters
+    ----------
+
+    image : `astropy.nddata.CCDData` or numpy array
+        Image containing the star.
+
+    center_guess : array or tuple
+        The position, in pixels, of the initial guess for the position of
+        the star. The coordinates should be horizontal first, then vertical,
+        i.e. opposite the usual Python convention for a numpy array.
+
+    cutout_size : int, optional
+        The default width of the cutout to use for finding the star.
+
+    max_iters : int, optional
+        Maximum number of iterations to go through in finding the center.
+
+    match_limit : int, optional
+        Maximum number of pixels to allow the COM centroid and Gaussian center to
+        differ.
+
+    Returns
+    -------
+
+    cen : array
+        The position of the star, in pixels, as found by the centroiding
+        algorithm.
+    """
+    pad = cutout_size // 2
+    x, y = center_guess
+
+    # Keep track of iterations
+    cnt = 0
+
+    # Grab the cutout...
+    sub_data = Cutout2D(image, center_guess, (cutout_size, cutout_size), mode="trim")
+    # ...do stats on it...
+    _, sub_med, _ = sigma_clipped_stats(sub_data.data)
+    # ...and centroid.
+    mask = (sub_data.data - sub_med) < 0
+    x_cm, y_cm = centroid_com(sub_data.data - sub_med, mask=mask)
+
+    # Translate centroid back to original image (maybe use Cutout2D instead)
```

Comment:
Yeah, maybe use Cutout2D instead... oh wait, you did, so you can remove the parenthetical comment... :)